### PR TITLE
[EASI-4386] - Homepage customization settings

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# EASI-4386

## Changes and Description

- Added view for selecting homepage settings
- Added view for selecting order of settings
- Added view for selection operational solutions
- Unit tests
- Cypress tests
- Feature flag around routes
- Success message

Data only persists on the solutions page (to save solutions) as well as the order page.  Selecting the settings on the settings page stores data in FE memory/router state.  Verify no mutation occurs on the settings page when you click 'Next', only on the order page when you select 'Save customizations'.

## How to test this change

- Navigate to `/homepage-setttings`
- Verify that `My Model Plans` is selected by default if base user
- Verify that `My Model Plans` (if collaborator) and `All Model Plans` selected is admin
- Verify that `Models with FFS CRs or TDLs` selected if FFS/Mac user
- Click selections, and select `Select solutions` on the relevant card
- In the dropdown, choose some solutions and save.  
- Verify selections persisted, solutions rendered on card.
- Make changes to setting selections if you wish and click next
- Change order, click back, forward, verify state persists.
- Click 'Save customizations' on order page, verify success message
- Go back and verify selections persist

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Comment by ClayBenson94** [comment]
@patrickseguraoddball Might be a question/clarification for UX, but wondering if we should either

1. Disallow/Disable the selection of operational solutions if you don't have the `Models using specific operational solutions` customization option selected
2. Automatically enable the `Models using specific operational solutions` option if you modify/save the solutions

I'd probably lean towards 1 (the link not being present/disabled when the checkbox isn't selected feels natural), but wanted to share!

Kapture 2024-06-27 at 12 38 53

File: src/i18n/en-US/home/settings.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      description:
+        'This section displays all of the Model Plans that you’re a team member of. Model links go to the Task List.'
+    },
+    ALL_MODEL_PLANS: {
+      heading: 'All Model Plans',
+      description:
+        'This searchable table displays all Model Plans (like the one on the Models tab).'
+    },
+    FOLLOWED_MODELS: {
+      heading: 'Models I’m following Models',
```
Comment (ClayBenson94) [inline src/i18n/en-US/home/settings.ts:23]:
```suggestion
      heading: 'Models I’m following',
```

File: src/i18n/en-US/home/settings.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  stepOne: '(Step 1 of 2)',
+  stepTwo: '(Step 2 of 2)',
+  description:
+    'Display the content that’s most meaningful to your role. You can update this at any time.',
+  descriptionTwo:
+    'Arrange the content however you wish. You can update this at any time.',
+  selection: 'Select which information to display on your homepage.',
+  selectionTwo: 'Set the display order for your homepage.',
+  settings: {
+    MY_MODEL_PLANS: {
```
Comment (ClayBenson94) [inline src/i18n/en-US/home/settings.ts:12]: Is there any way we can type this `settings` object easily? It'd be nice to ensure we're covering all cases/using the enums from the generated GQL for these

If not, definitely not a dealbreaker!

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]

**Comment by patrickseguraoddball** [comment]
Addressed the changes but waiting to hear back from UX

**Review by ClayBenson94 — APPROVED** [review]

**Pull request merged**